test/e2e/tests/signature-request.spec.js

```diff
@@ -210,20 +345,70 @@ describe('Sign Typed Data Signature Request', function () {
           windowHandles,
         );
 
-        const title = await driver.findElement(
-          '.request-signature__content__title',
+        await verifyAndAssertSignTypedData(driver);
+
+        // Approve signing typed data
+        await driver.clickElement({ text: 'Sign', tag: 'button' });
+        await driver.waitUntilXWindowHandles(2);
+        windowHandles = await driver.getAllWindowHandles();
+
+        // switch to the Dapp and verify the signed addressed
+        await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
+        await driver.clickElement('#signTypedDataVerify');
+        const recoveredAddress = await driver.findElement(
+          '#signTypedDataVerifyResult',
         );
-        const origin = await driver.findElement('.request-signature__origin');
-        const message = await driver.findElements(
-          '.request-signature__row-value',
+        assert.equal(await recoveredAddress.getText(), publicAddress);
+      },
+    );
+  });
+
+  it('can queue multiple Signature Request and confirm', async function () {
```

Comment:
Is there much difference between the V3 and V4 tests, would it be worth using a single template function with arguments for the validation methods perhaps?

I wonder if we could even abstract something for all of the signature types to remove as much duplication as possible across all three files?

Reply:
I got to a point where I managed to abstract between V1, V3 and V4 but when I tried to expand to other signings it got fragile. I could give it another try.